crates/core/component/auction/src/auction/nft.rs

```diff
@@ -1,15 +1,31 @@
 use crate::auction::id::AuctionId;
 use anyhow::{anyhow, Result};
+use penumbra_asset::asset::{self};
 use penumbra_proto::{core::component::auction::v1alpha1 as pb, DomainType};
 
 /// An non-fungible token (NFT) tracking the state and ownership of an auction.
 #[derive(Debug, Clone)]
 pub struct AuctionNft {
     /// The unique identifier for the auction this nft resolves to.
-    id: AuctionId,
+    pub id: AuctionId,
     /// The state of an auction, its specific semantics depend on the
     /// type of auction the NFT resolves to.
-    seq: u64,
+    pub seq: u64,
+    /// The metadata corresponding to the nft denom.
+    pub metadata: asset::Metadata,
+}
+
+impl AuctionNft {
+    pub fn new(id: AuctionId, seq: u64) -> AuctionNft {
+        let metadata = asset::REGISTRY
+            .parse_denom(&format!("auctionnft_{seq}_{id}"))
+            .expect("auction nft denom is valid");
```

Comment:
I think I want to do a second pass on both auction and lp nfts to make their constructor fallible.